include/pybind11/detail/native_enum_data.h

```diff
@@ -67,10 +78,11 @@ class native_enum_data {
     std::type_index enum_type_index;
 
 private:
-    object parent_scope;
+    handle parent_scope;
```

Comment:
What is keeping the parent scope alive? Could you please add a comment to explain?

Reply:
native_enum_data only lives for the duration of a single `py::native_enum` binding statement, so I assumed the arguments to that statement will live longer than the native_enum_data. It is technically possible though to stash the `native_enum` object aside and go back to add more enumerators piecemeal, at which point this could theoretically dangle, so I switched it back to `object`.

The constructor parameter should still be `handle` so that you can write things like `py::native_enum<T>(mod.attr("SomeClass"), ...)`